AddNewRecipes/Program.cs

```diff
@@ -45,39 +46,39 @@ public static async Task<int> Main(string[] args)
         public static void RunPatch(IPatcherState<ISkyrimMod, ISkyrimModGetter> state)
         {
             List<IngrCombination> combinations = new List<IngrCombination>();
-            IEnumerable<IIngredientGetter> ingredients = state.LoadOrder.PriorityOrder.OnlyEnabled().Ingredient().WinningOverrides();
-            ingredients = from ingrs in ingredients where !SkipPlugins.Contains(ingrs.FormKey.ModKey.Name.ToLower()) select ingrs;
-            ingredients = from ingrs in ingredients where (!SkipIngrs.Intersect(ingrs.Name?.ToString()?.Split()!).Any() || SkipIngrs.Contains(ingrs.Name?.ToString())) select ingrs;
-            ingredients = from ingrs in ingredients where !String.IsNullOrEmpty(ingrs.Name?.String) select ingrs;
-            IEnumerator<IIngredientGetter> enumerator = ingredients.GetEnumerator();
+            var ingredients = state.LoadOrder.PriorityOrder.OnlyEnabled().Ingredient().WinningOverrides()
+                .Where(x => !SkipPlugins.Contains(x.FormKey.ModKey.Name.ToLower()))
+                .Where(x => (!SkipIngrs.Intersect(x.Name?.ToString()?.Split()!).Any() || SkipIngrs.Contains(x.Name?.ToString())))
+                .Where(x => !String.IsNullOrEmpty(x.Name?.String))
+                .ToList();
```

Comment:
The big addition here is the ToList.  Without it, every time ingredients was accessed (again n ^ 3 times), it was going to disk, reading in the bytes, parsing all the ingredients from the bytes, then giving them to you.  Every time.   This ToList addition means that work is only done once, so downstream users don't have to reparse the same ingredients over and over.